packages/services/rate-limit/src/router.ts

```diff
@@ -0,0 +1,57 @@
+import { createLogger, reportReadiness } from '@hive/service-common';
+import { createConnectionString } from '@hive/storage';
+import { createFetchAPIHandler } from '@valu/trpc-fetch-api-adapter';
+import { createRouter } from '@whatwg-node/router';
+import { rateLimitApiRouter } from './api';
+import { env } from './environment';
+import { createRateLimiter } from './limiter';
+
+const rateLimitRouter = createRouter();
+
+const logger = createLogger(env.log.level);
+
+const ctx = createRateLimiter({
+  logger,
+  rateLimitConfig: {
+    interval: env.limitCacheUpdateIntervalMs,
+  },
+  rateEstimator: env.hiveServices.usageEstimator,
+  emails: env.hiveServices.emails ?? undefined,
+  storage: {
+    connectionString: createConnectionString(env.postgres),
+  },
+});
+
+const respondWithTRPC = createFetchAPIHandler({
+  router: rateLimitApiRouter,
+  createContext: async (): Promise<any> => ctx,
+});
+
+rateLimitRouter.all('/trpc/:path+', async (req: any) => {
```

Comment:
```suggestion
rateLimitRouter.all('/trpc/:path+', responseWithTRPC)
```
We can pass this handler without an additional `async/await`, no?